File: ui/src/test/java/org/wcl/mfn/ui/integration/uielements/pageobjectmodel/HomePageModel.java
package org.wcl.mfn.ui.integration.uielements.pageobjectmodel;

import org.openqa.selenium.*;
import org.wcl.mfn.ui.config.common.NavigationBarConfig;


public class HomePageModel {

    private WebDriver driver;
    private NavigationBarConfig navigationBarConfig;

    public HomePageModel(WebDriver driver, NavigationBarConfig navigationBarConfig) {
        this.driver = driver;
        this.navigationBarConfig = navigationBarConfig;
    }

    public WebElement navigationBar() {
        return driver.findElements(By.id(navigationBarConfig.navigationBarId())).getFirst();
    }

    public String homeLink() {
        return driver.findElements(By.id(navigationBarConfig.navigationBarHomeId())).getFirst()
                .getAttribute("href");
    }

    public String contractCalculatorLink() {
        return driver.findElements(By.id(navigationBarConfig.navigationBarContractCalculatorId())).getFirst()
                .getAttribute("href");
    }

    public String title() {
        return driver.getTitle();
    }

}

File: ui/src/test/kotlin/org/wcl/mfn/ui/integration/uielements/pageobjectmodel/HomePageModel.kt
package org.wcl.mfn.ui.integration.uielements.pageobjectmodel

import org.openqa.selenium.*
import org.wcl.mfn.config.ui.common.NavigationBarConfig

class HomePageModel(private val driver: WebDriver, private val navigationBarConfig: NavigationBarConfig) {
    fun navigationBar(): WebElement {
        return driver.findElements(By.id(navigationBarConfig.navigationBarId())).first()
    }

    fun homeLink(): String {
        return driver.findElements(By.id(navigationBarConfig.navigationBarHomeId())).first()
            .getAttribute("href")
    }

    fun contractCalculatorLink(): String {
        return driver.findElements(By.id(navigationBarConfig.navigationBarContractCalculatorId())).first()
            .getAttribute("href")
    }

    val title: String
        get() = driver.title
}
